src/components/TransactionHistoryRows/TransactionGenericRow.tsx

```diff
@@ -108,43 +108,20 @@ export const TransactionGenericRow = ({
   } = GetTxLayoutFormats({ parentWidth })
 
   const transfers = useMemo(() => {
-    return displayTransfers.map((transfer, index) => (
-      <Stack
-        alignItems='center'
-        key={index}
-        flex={1}
-        mt={{ base: 2, md: 0, xl: compactMode ? 2 : 0 }}
-        direction={index === 0 ? 'row' : 'row-reverse'}
-        textAlign={index === 0 ? 'left' : 'right'}
-      >
-        <AssetIcon
-          assetId={transfer.asset?.assetId}
-          boxSize={{ base: '24px', lg: compactMode ? '24px' : '40px' }}
-        />
-        <Box flex={1}>
-          <Amount.Crypto
-            color='inherit'
-            fontWeight='medium'
-            value={fromBaseUnit(transfer.value, transfer.asset?.precision ?? FALLBACK_PRECISION)}
-            symbol={transfer.asset?.symbol ?? FALLBACK_SYMBOL}
-            maximumFractionDigits={4}
-          />
-          {transfer.marketData.price && (
-            <Amount.Fiat
-              color='gray.500'
-              fontSize='sm'
-              lineHeight='1'
-              value={bnOrZero(
-                fromBaseUnit(transfer.value, transfer.asset?.precision ?? FALLBACK_PRECISION),
-              )
-                .times(transfer.marketData.price)
-                .toString()}
-            />
-          )}
-        </Box>
-      </Stack>
-    ))
-  }, [compactMode, displayTransfers])
+    return displayTransfers.map((transfer, index) => {
+      const displayTransferType = transfer.type
+      const typeTxTransfers = (txTransfers ?? []).filter(({ type }) => type === displayTransferType)
+      const hasManyTypeTransfers = typeTxTransfers.length > 1
+
+      if (hasManyTypeTransfers) {
+        return (
+          <AssetsTransfers index={index} compactMode={compactMode} transfers={typeTxTransfers} />
+        )
+      }
+
+      return <AssetTransfer index={index} compactMode={compactMode} transfer={transfer} />
```

Comment:
Use a functional single return here, sir?

We should use multiple return points with caution, as a lack of caution is how we end up with spaghetti components that require the entire component function to be loaded into mental memory before it can be understood, 
à la [`packages/swapper/src/swappers/thorchain/utils/getPriceRatio/getPriceRatio.ts`](https://github.com/shapeshift/lib/blob/main/packages/swapper/src/swappers/thorchain/utils/getPriceRatio/getPriceRatio.ts).

```suggestion
      return hasManyTypeTransfers ? (
        <AssetsTransfers index={index} compactMode={compactMode} transfers={typeTxTransfers} />
      ) : (
        <AssetTransfer index={index} compactMode={compactMode} transfer={transfer} />
      )
```